File: WebSocketChat.Frontend/src/app/shared/models/chat/messages/user-identifier-message.ts
import { ChatUser } from './../chat-user';
import { Message, MessageIdentifier } from '../message';
import { MessageType } from '../message-type';

export class UserIdentifierMessage extends MessageIdentifier implements Message {
	messageType : MessageType = MessageType.UserIdentifier;

	/**Indicates whether the user to whom this message is send to is the one who caused this message (he joined the server, changed his nickname etc.) */
	isOriginOfMessage: boolean;

	/**The user that has joined the room/server */
	chatUser: ChatUser;
}

File: WebSocketChat.Server/WebSocketChat.Server/Models/Chat/Messages/UserIdentifierMessage.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace WebSocketChat.Server.Models.Chat.Messages
{
    public class UserIdentifierMessage : Message
    {
        public UserIdentifierMessage()
        {
            MessageType = MessageType.UserIdentifier;
            Timestamp = DateTime.Now;
        }

        /// <summary>
        /// Specifies when the identification did happen (when did the user has joined the server etc.)
        /// </summary>
        public DateTime Timestamp { get; set; }

        /// <summary>
        /// Indicates whether the user to whom this message is send to is the one who caused this message (he joined the server, changed his nickname etc.)
        /// </summary>
        public bool IsOriginOfMessage { get; set; }

        /// <summary>
        /// The user that has joined the room/server
        /// </summary>
        public ChatPartner ChatUser { get; set; }
    }
}
